R/teal_data-class.R

```diff
@@ -53,18 +54,11 @@ setClass(
 #' @param code (`character` or `language`) code to reproduce the `data`.
 #'   Accepts and stores comments also.
 #' @param join_keys (`join_keys`) object
-#' @param datanames (`character`) names of datasets passed to `data`.
-#'   Needed when non-dataset objects are needed in the `env` slot.
 #' @rdname new_teal_data
 #' @keywords internal
-new_teal_data <- function(data,
-                          code = character(0),
-                          join_keys = join_keys(),
-                          datanames = names(data)) {
+new_teal_data <- function(data, code = character(0), join_keys = join_keys()) {
```

Comment:
Let's deprecate and not remove argument for now. 
```suggestion
new_teal_data <- function(data, code = character(0), join_keys = join_keys(), datanames = lifecycle::deprecated()) {
```